zebra/zebra_nhg.h

```diff
@@ -24,6 +24,245 @@
 #define __ZEBRA_NHG_H__
 
 #include "zebra/rib.h"
+#include "lib/nexthop_group.h"
+
+#include "zebra/zebra_dplane.h"
+
+/* This struct is used exclusively for dataplane
+ * interaction via a dataplane context.
+ *
+ * It is designed to mimic the netlink nexthop_grp
+ * struct in include/linux/nexthop.h
+ */
+struct nh_grp {
+	uint32_t id;
+	uint8_t weight;
+};
+
+PREDECL_RBTREE_UNIQ(nhg_connected_tree);
+
+struct nhg_hash_entry {
```

Comment:
In the top of zebra router it has this:

```
/*                                                                               
 * This header file contains the idea of a router and as such                    
 * owns data that is associated with a router from zebra's                       
 * perspective.                                                                  
 */                                                                              
```

Since that is where the route tables are kept as well I felt its the appropriate place for nexthop group hashes.

The rb tree you are referencing here are internal ones only used from `nhg -> nhg` via the depend/dependent relationship.

But yes, I think a comment is probably owed there.